### ISSUE TITLE
Ask for Tworoom dataset training config

### ISSUE THREAD

**Issue by YuCYstar** [galilai-group/stable-worldmodel#248]
Thank you very much for your great work. I really appreciate this valuable contribution.

I am currently trying to reproduce the TwoRoom results reported in the paper using LeWorldModel. I followed the hyperparameters described in the paper, together with the default settings provided in the GitHub repository. However, when training on TwoRoom, I can only achieve around a 48% success rate, which is significantly lower than the result reported in the paper.

Here is my current training configuration:

```python
# ============================================================================
# ALL HYPERPARAMETERS LIVE HERE — paper-strict for TwoRoom
# ============================================================================
CONFIG = {
    # ---- Run identity ----
    'output_model_name': 'TwoRoom/lewm_tworoom_single',
    'subdir':            'TwoRoom/lewm_tworoom_single',
    'seed':              3072,

    # ---- Trainer ----
    'max_epochs':        10,
    'precision':         'bf16',
    'gradient_clip_val': 1.0,
    'devices':           1,

    # ---- DataLoader ----
    'batch_size_per_device': 128,
    'num_workers':           6,
    'train_split':           0.9,

    # ---- Optimizer ----
    'optimizer_type': 'AdamW',
    'lr':             5e-5,
    'weight_decay':   1e-3,

    # ---- Scheduler ----
    # spt smart default automatically uses trainer.estimated_stepping_batches as max_steps
    'scheduler_type': 'LinearWarmupCosineAnnealingLR',

    # ---- World Model (TwoRoom paper-strict — history_size=1 is the only difference) ----
    'history_size': 1,                 # Appendix D: TwoRoom=1
    'num_preds':    1,
    'img_size':     224,
    'patch_size':   14,
    'embed_dim':    192,
    'encoder_size': 'tiny',

    # ---- Loss ----
    'sigreg_weight':   0.09,
    'sigreg_knots':    17,
    'sigreg_num_proj': 1024,

    # ---- Predictor (AR transformer) ----
    'pred_depth':       6,
    'pred_heads':       16,
    'pred_mlp_dim':     2048,
    'pred_dim_head':    64,
    'pred_dropout':     0.1,
    'pred_emb_dropout': 0.0,

    # ---- Projector / pred_proj ----
    'proj_hidden_dim': 2048,

    # ---- Data (TwoRoom) ----
    'dataset_name':   'tworoom.h5',
    'frameskip':      5,
    'keys_to_load':  ['pixels', 'action', 'proprio'],
    'keys_to_cache': ['action', 'proprio'],
}
# ============================================================================
```

Would it be possible for you to provide a more detailed training script or the exact configuration used for the TwoRoom experiments in the paper? In particular, I would be very grateful if you could clarify whether there are any additional settings related to data preprocessing, evaluation protocol, training epochs, scheduler details, random seeds, or checkpoint selection that are important for reproducing the reported results.

Thank you again for your time and for releasing this work. Any guidance would be greatly appreciated.

**Comment by lucas-maes** [comment]
Hi! ty for the message. I think the confusing part is that we set the lr scheduler for 100 epochs, but stop training after 10!

That's most likely the issue! Make sure that this is what your code is also doing!

Best,
Lucas
